ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/websocket/subscription/SubscriptionManager.java

```diff
@@ -158,4 +161,20 @@ public void sendMessage(final Long subscriptionId, final JsonRpcResult msg) {
           }
         });
   }
+
+  @Override
+  public void onPrivateTransactionProcessed(final PrivateTransactionEvent event) {
+    // When a user is removed from a privacy group, remove all subscriptions from that user to that
+    // group
+    subscriptionsOfType(SubscriptionType.LOGS, PrivateLogsSubscription.class).stream()
+        .filter(
+            subscription ->
+                subscription.getEnclavePublicKey().equals(event.getEnclavePublicKey())
+                    && subscription.getPrivacyGroupId().equals(event.getPrivacyGroupId()))
+        .forEach(
+            subscription ->
+                this.unsubscribe(
+                    new UnsubscribeRequest(
+                        subscription.getSubscriptionId(), subscription.getConnectionId())));
```

Comment:
do we have to worry about this request failing?

Reply:
I don't believe so. The fail case is if the specified connection doesn't exist, which shouldn't occur given we're retrieving those details from the current subscription map.